cirq-google/cirq_google/engine/abstract_job.py

```diff
@@ -162,25 +164,31 @@ def delete(self) -> Optional[bool]:
         """Deletes the job and result, if any."""
 
     @abc.abstractmethod
-    def batched_results(self) -> Sequence[Sequence[EngineResult]]:
+    async def batched_results_async(self) -> Sequence[Sequence[EngineResult]]:
         """Returns the job results, blocking until the job is complete.
 
         This method is intended for batched jobs.  Instead of flattening
         results into a single list, this will return a List[Result]
         for each circuit in the batch.
         """
 
+    batched_results = duet.sync(batched_results_async)
```

Comment:
This threw me for a loop when I first read it. Could we instead use something more standard, like
```python
def batched_results(self) -> Sequence[Sequence[EngineResult]]:
    """Synchronous wrapper around batched_results_async."""
    return duet.sync(self.batched_results_async())
```
or would that have different behavior? Same for the other two.

Reply:
`duet.sync` works like a decorator, so it would have to be `duet.sync(self.batched_results_async)()`, but in that case one would just run the async method directly: `duet.run(self.batched_results_async)`. That said, `duet.sync` was specifically designed for this way of wrapping async functions without having to repeat the signatures, to reduce boilerplate and the requirement to keep two copies of the signatures synchronized when making changes. If we decide not to use this we will want to change existing usage of `duet.sync`, for example numerous usages in [engine_client.py](https://github.com/quantumlib/Cirq/blob/master/cirq-google/cirq_google/engine/engine_client.py#L219).

Reply:
Note also that methods wrapped in this way still get handled correctly by the doc generators, for example see docs for [`EngineProgram.list_jobs`](https://quantumai.google/reference/python/cirq_google/engine/EngineProgram#list_jobs).